ingestion/src/levels.rs

```diff
@@ -0,0 +1,102 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+use thiserror::Error;
+use tokio_postgres::NoTls;
+use tracing::error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("postgres returned an error: {0}")]
+    Database(#[from] tokio_postgres::Error),
+    #[error("RwLock was poisoned: {0}")]
+    Lock(String),
+}
+
+#[derive(Debug, Clone)]
+pub struct Level {
+    hlevel: i32,
+    hlevel_scale: i32,
+}
+
+#[cfg(feature = "integration_tests")]
+impl Level {
+    pub fn new(hlevel: i32, hlevel_scale: i32) -> Level {
+        Level {
+            hlevel,
+            hlevel_scale,
+        }
+    }
+}
+
+type ParamID = i32;
+
+pub type ParamLevelTable = HashMap<ParamID, Level>;
+
+/// Get a fresh cache of levels from stinfosys
+pub async fn fetch_levels(stinfo_conn_string: &str) -> Result<ParamLevelTable, Error> {
+    // get stinfo conn
+    let (client, conn) = tokio_postgres::connect(stinfo_conn_string, NoTls).await?;
+
+    // conn object independently performs communication with database, so needs it's own task.
+    // it will return when the client is dropped
+    tokio::spawn(async move {
+        if let Err(e) = conn.await {
+            error!("connection error: {}", e);
+        }
+    });
+
+    // query param table
+    let rows = client
+        .query(
+            "SELECT standard_hlevel, hlevel_scale, paramid 
+                FROM param WHERE standard_hlevel is not null",
+            &[],
+        )
+        .await?;
+
+    // build hashmap of param permits
+    let mut param_level = HashMap::new();
+
+    for row in rows {
+        let hlevel_scale: i32 = row.get(1);
+        if ![0, -2].contains(&hlevel_scale) {
+            // currently only have 0 and -2, aka m and cm
+            param_level.insert(
+                row.get(3),
+                Level {
+                    hlevel: row.get(0),
+                    hlevel_scale,
+                },
+            );
+        }
```

Comment:
I would remove this if condition and add a runtime warning/error if the level is not 0 or -2 instead (in `param_get_level`? Otherwise I we should log here the params with weird scales?

```suggestion
        param_level.insert(
            row.get(2),
            Level {
                hlevel: row.get(0),
                hlevel_scale: row.get(1),
            },
        );
```

Reply:
Disagree! I think it makes more sense to catch these when we load from stinfosys, not when we handle a datum. With the change we will fail to ingest data that uses it, which I don't think is ideal, and we lose our chance to detect the problem before that happens.

I agree that we should `error!` log and maybe have a metric too though

Reply:
Given we eventually won't have kvalobs to fall back on, I think it's better to set the precedent of ingesting something that needs fixing, rather than risk missing the data entirely. We will need to do corrections to data based on stinfosys changes and other mistakes anyway, which falls to the content managers

Reply:
Seems reasonable to me to only check at load time, and exclude unsupported entries from the map.

That way the data will have lvl = `NULL` in `labels.met` and their original hlevel in `labels.kvalobs`/`labels.obsinn`.

This should be easy to find a fix, because affected timeseries will have `NULL` in `labels.met` and `NOT NULL` in their source specific label, that can only be caused by this. Not to mention we should know what is affected simply because we know what the new param is in stinfosys, and can just look for timeseries with that paramid.

Reply:
Changed back as I saw fit based on this conversation in recent commit